addons/stock/report/report_stock_reception.py

```diff
@@ -0,0 +1,283 @@
+        """ Links picking move(s) to the next step (i.e. move_ids in another picking)
+        """
+        outs = self.env['stock.move'].browse(move_ids)
+        # Split outs with only part of demand assigned to prevent reservation problems later on.
+        # We do this first so we can create their split moves in batch
+        out_to_new_out = OrderedDict()
+        new_move_vals = []
+        for out, qty_to_link in zip(outs, qtys):
+            if float_compare(out.product_qty, qty_to_link, precision_rounding=out.product_id.uom_id.rounding) == 1:
+                new_move_vals += out._split(out.product_qty - qty_to_link)
+                out_to_new_out[out.id] = self.env['stock.move']
+        new_outs = self.env['stock.move'].create(new_move_vals)
+        new_outs._action_confirm(merge=False)
+        i = 0
+        for k, dummy in out_to_new_out.items():
+            out_to_new_out[k] = new_outs[i]
+            i += 1
+
+        for out, qty_to_link, ins in zip(outs, qtys, in_ids):
+            potential_ins = self.env['stock.move'].browse(ins)
+            if out.id in out_to_new_out:
+                new_out = out_to_new_out[out.id]
+                if potential_ins[0].state != 'done' and out.reserved_availability:
+                    # let's assume if 1 of the potential_ins isn't done, then none of them are => we are only assigning the not-reserved
+                    # qty and the new move should have all existing reserved quants (i.e. move lines) assigned to it
+                    out.move_line_ids.move_id = new_out
+                elif potential_ins[0].state == 'done' and out.reserved_availability > qty_to_link:
+                    # let's assume if 1 of the potential_ins is done, then all of them are => we can link them to already reserved moves, but we
+                    # need to make sure the reserved qtys still match the demand amount the move (we're assigning).
+                    out.move_line_ids.move_id = new_out
+                    assigned_amount = 0
+                    for move_line_id in new_out.move_line_ids:
+                        if assigned_amount + move_line_id.product_qty > qty_to_link:
+                            new_move_line = move_line_id.copy({'product_uom_qty': 0, 'qty_done': 0})
+                            move_line_id.product_uom_qty = out.product_id.uom_id._compute_quantity(qty_to_link - assigned_amount, out.product_uom, rounding_method='HALF-UP')
+                            new_move_line.product_uom_qty -= out.product_id.uom_id._compute_quantity(move_line_id.product_qty, out.product_uom, rounding_method='HALF-UP')
+                        move_line_id.move_id = out
+                        assigned_amount += move_line_id.product_qty
+                        if float_compare(assigned_amount, qty_to_link, precision_rounding=out.product_id.uom_id.rounding) == 0:
+                            break
+                (out | new_out)._compute_reserved_availability()
```

Comment:
It is mandatory to call the compute directly, the depends doesn't make the job (`move_line_ids.product_qty`) ?